### DISCUSSION TITLE
Can't pip install on Raspberry Pi 4b w/ Raspbian 10 Buster

### DISCUSSION THREAD

**Discussion by jsmif** [hbldh/bleak#1339] [Q&A]
* bleak version: N/A
* Python version: 3.7.3
* Operating System: 
```
cat /etc/os-release
PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
```
* BlueZ version (`bluetoothctl -v`) in case of Linux: 5.66 (manually installed from source just now)

### Description

I'm trying to install bleak on a raspberry pi since it doesn't work on Ubuntu 22.04 #1333 

### What I Did

```
pi@pi1:~/Downloads/bleak $ pip install -e .
Directory '.' is not installable. File 'setup.py' not found.
pi@pi1:~/Downloads/bleak $ pip3 install -e .
Directory '.' is not installable. File 'setup.py' not found.
pip install bleak
Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
Collecting bleak
  Downloading https://files.pythonhosted.org/packages/15/69/3f608bf39305d9c2dceaa58076b3e40ed308adeeff2b6f27325ab0805624/bleak-0.16.0-py2.py3-none-any.whl (125kB)
    100% |████████████████████████████████| 133kB 2.1MB/s 
Collecting dbus-next; platform_system == "Linux" (from bleak)
  Could not find a version that satisfies the requirement dbus-next; platform_system == "Linux" (from bleak) (from versions: )
No matching distribution found for dbus-next; platform_system == "Linux" (from bleak)
```

I suspect this might be because this is a 32-bit system?

**Comment by jsmif** [comment]
Bah! I wanted bug #1337!

**Comment by dlech** [comment]
If you are not in a virtual environment, you need to use `pip3` instead of `pip` so that it searches for Python 3 packages instead of Python 2.

**Comment by jsmif** [comment]
Bah, the one of 4 options I didn't try! Yes `pip3 install bleak` worked.

Then it gives a similar, but slightly different, error to #1333:

```
python3 service_explorer.py --address 6C:4A:85:2C:C3:A9
2023-06-15 02:07:17,611 __main__ INFO: starting scan...
2023-06-15 02:07:18,102 __main__ INFO: connecting to device...
Traceback (most recent call last):
  File "service_explorer.py", line 129, in <module>
    asyncio.run(main(args))
  File "/usr/lib/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "service_explorer.py", line 43, in main
    services=args.services,
  File "/home/pi/.local/lib/python3.7/site-packages/bleak/__init__.py", line 491, in __aenter__
    await self.connect()
  File "/home/pi/.local/lib/python3.7/site-packages/bleak/__init__.py", line 531, in connect
    return await self._backend.connect(**kwargs)
  File "/home/pi/.local/lib/python3.7/site-packages/bleak/backends/bluezdbus/client.py", line 244, in connect
    assert_reply(reply)
  File "/home/pi/.local/lib/python3.7/site-packages/bleak/backends/bluezdbus/utils.py", line 22, in assert_reply
    raise BleakDBusError(reply.error_name, reply.body)
bleak.exc.BleakDBusError: [org.bluez.Error.Failed] br-connection-profile-unavailable
```

**Reply by dlech** [reply]
We would need to see Bluetooth packet logs to be able to see what might be going on.

**Comment by thijstriemstra** [comment]
Have you tried on raspberry pi os bullseye? buster is quite old.

**Comment by iosifnicolae2** [comment]
We're experiencing the same error, I've attached below the source code and some logs:
https://gist.github.com/iosifnicolae2/bc70134f317ce24234fa58740272fc4c

If you have any idea on how to fix this, thank you in advance :)

PS. The code works just fine on MacOS, but when running on Raspberry PI 3 A+, we're getting the `bleak.exc.BleakDBusError: [org.bluez.Error.NotAvailable] br-connection-profile-unavailable` error.
We're trying to connect to `Bosch glm50c` by adapting this code (https://github.com/ketan/Bosch-GLM50C-Rangefinder/blob/main/python/main.py).

**Reply by iosifnicolae2** [reply]
Tomorrow, I would like to try to ignore this error.

PS. Also, when we try to connect to the BLE device from Raspberry PI interface, we're getting the same error.

```
bluetoothctl --version
bluetoothctl: 5.66
```
